File: src/Atlasd/Battlenet/Exceptions/GameProtocolException.cs
using System;

namespace Atlasd.Battlenet.Exceptions
{
    class GameProtocolException : ProtocolException
    {
        public GameProtocolException(ClientState client) : base(Battlenet.ProtocolType.Types.Game, client) { }
        public GameProtocolException(ClientState client, string message) : base(Battlenet.ProtocolType.Types.Game, client, message) { }
        public GameProtocolException(ClientState client, string message, Exception innerException) : base(Battlenet.ProtocolType.Types.Game, client, message, innerException) { }
    }
}

File: src/AtlasV/Battlenet/Exceptions/GameProtocolException.vb
Imports System

Namespace AtlasV.Battlenet.Exceptions
    Class GameProtocolException
        Inherits ProtocolException

        Public Sub New(ByVal varClient As ClientState)
            MyBase.New(Battlenet.ProtocolType.Types.Game, varClient)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String)
            MyBase.New(Battlenet.ProtocolType.Types.Game, varClient, varMessage)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String, ByVal varInnerException As Exception)
            MyBase.New(Battlenet.ProtocolType.Types.Game, varClient, varMessage, varInnerException)
        End Sub
    End Class
End Namespace
